File: src/web3/js/signing.js
import crypto from "crypto";

export class DeterministicSigner {
  static sign(message) {
    const hash = crypto.createHash("sha256")
      .update(message)
      .digest("hex");

    return {
      message,
      signature: hash,
      algorithm: "sha256",
      deterministic: true
    };
  }
}

File: src/web3/python/signing.py
import hashlib

class DeterministicSigner:
    @staticmethod
    def sign(message: str):
        hashed = hashlib.sha256(message.encode()).hexdigest()
        return {
            "message": message,
            "signature": hashed,
            "algorithm": "sha256",
            "deterministic": True
        }
